### PR TITLE
Add accessibility labels

### PR DESCRIPTION
**Pull request by gunnigylfa**
In an effort to make our reason react native apps accessible for as many as possible we added:
- Accessibility label prop support for `TouchableHighlight`, `TouchableOpacity`, `TouchableWithoutFeedback` and `Text`. 
- Changed the type of accessibilityLabel in `View` from  `ReasonReact.reactElement` to string.

### CODE REVIEW

File: src/components/text.rei [hunk 1]
```diff
@@ -2,6 +2,7 @@ module type TextComponent = {
   let make:
     (
       ~accessible: bool=?,
+      ~accessibilityLabel: string=?,
```
Comment (kristinns) [inline src/components/text.rei]: I would remove `accessibilityLabel` from the `Text` element, it didn't seem to work when I tested it. 
I think if you need it, you should wrap it with `View` and add `accessibilityLabel` on that instead.
Also, it's not mentioned in the [documentation](https://facebook.github.io/react-native/docs/accessibility.html)

Comment (gunnigylfa) [reply src/components/text.rei]: Nice catch! removed it removed in: ef88999

File: lib/js/src/components/text.js [hunk 2]
```diff
@@ -55,9 +56,10 @@ function CreateComponent(Impl) {
 
 var Impl = /* module */[/* Text */ReactNative.Text];
 
-function make(accessible, allowFontScaling, ellipsizeMode, numberOfLines, onLayout, onLongPress, onPress, pressRetentionOffset, selectable, style, testID, selectionColor, textBreakStrategy, adjustsFontSizeToFit, minimumFontScale, suppressHighlighting, value, children) {
+function make(accessible, accessibilityLabel, allowFontScaling, ellipsizeMode, numberOfLines, onLayout, onLongPress, onPress, pressRetentionOffset, selectable, style, testID, selectionColor, textBreakStrategy, adjustsFontSizeToFit, minimumFontScale, suppressHighlighting, value, children) {
```
Comment (wokalski) [inline lib/js/src/components/text.js]: You forgot to build after removing `accessibilityLabel` from `Text`. The rest is awesome. Good job

Comment (gunnigylfa) [reply lib/js/src/components/text.js]: Ah nice catch, spaced out on that one. Fixed in: d0ccfc1

**Comment by wokalski** [comment]
@gunnigylfa good job!

**Comment by wokalski** [comment]
@kristinns @MoOx thanks for reviewing

**Pull request merged**